src/threadwrappers.cpp

```diff
@@ -223,19 +109,46 @@ pthread_create(pthread_t *thread,
    */
   bool threadCreationLockAcquired = ThreadSync::threadCreationLockLock();
   ThreadSync::incrementUninitializedThreadCount();
-  retval = _real_pthread_create(thread, attr, pthread_start, threadArg);
+
+  curThread->processingPthreadCreate = true;
+  retval = _real_pthread_create(pth, attr, pthread_start, thread);
+  curThread->processingPthreadCreate = false;
+
   if (threadCreationLockAcquired) {
     ThreadSync::threadCreationLockUnlock();
   }
+
   if (retval == 0) {
-    ProcessInfo::instance().clearPthreadJoinState(*thread);
+    ProcessInfo::instance().clearPthreadJoinState(*pth);
   } else { // if we failed to create new pthread
-    JALLOC_HELPER_FREE(threadArg);
     ThreadSync::decrementUninitializedThreadCount();
   }
   return retval;
 }
 
+// Make sure __clone is not called without pthread_create.
+extern "C" int
+__clone(int (*fn)(void *arg),
+        void *child_stack,
+        int flags,
+        void *arg,
+        int *parent_tidptr,
+        struct user_desc *newtls,
+        int *child_tidptr)
+{
```

Comment:
Let's add a FIXME comment here:
    `// FIXME:  Replace this with __clone_standalone_wrapper()`

By the way, note that `src/miscwrappers.cpp` and maybe other places.  But let's leave that in the code if we're agreed on replacing this with new wrappers for `__clone()`.